src/mochaAvoidOnlyRule.ts

```diff
@@ -25,38 +26,46 @@ export class Rule extends Lint.Rules.AbstractRule {
-        if (MochaUtils.isMochaTest(node)) {
-            super.visitSourceFile(node);
-        }
-    }
+function walk(ctx: Lint.WalkContext<void>) {
+    function cb(node: ts.Node): void {
+        if (tsutils.isCallExpression(node)) {
+            if (tsutils.isPropertyAccessExpression(node.expression)) {
+                if (node.arguments.length === 2) {
+                    if (tsutils.isStringLiteral(node.arguments[0])) {
+                        if (tsutils.isFunctionExpression(node.arguments[1]) || tsutils.isArrowFunction(node.arguments[1])) {
+                            const text = node.expression.getText();
+                            switch (text) {
+                                case 'it.only':
+                                    ctx.addFailureAt(node.getStart(), text.length, Rule.FAILURE_STRING_IT);
+                                    break;
+
+                                case 'specify.only':
+                                    ctx.addFailureAt(node.getStart(), text.length, Rule.FAILURE_STRING_SPECIFY);
+                                    break;
+
+                                case 'describe.only':
+                                    ctx.addFailureAt(node.getStart(), text.length, Rule.FAILURE_STRING_DESCRIBE);
+                                    break;
 
-    protected visitCallExpression(node: ts.CallExpression): void {
-        if (node.expression.kind === ts.SyntaxKind.PropertyAccessExpression) {
-            if (node.arguments.length === 2) {
-                if (node.arguments[0].kind === ts.SyntaxKind.StringLiteral) {
-                    if (
-                        node.arguments[1].kind === ts.SyntaxKind.FunctionExpression ||
-                        node.arguments[1].kind === ts.SyntaxKind.ArrowFunction
-                    ) {
-                        if (node.expression.getText() === 'it.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_IT);
-                        } else if (node.expression.getText() === 'specify.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_SPECIFY);
-                        } else if (node.expression.getText() === 'describe.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_DESCRIBE);
-                        } else if (node.expression.getText() === 'context.only') {
-                            this.addFailureAt(node.getStart(), node.expression.getText().length, Rule.FAILURE_STRING_CONTEXT);
+                                case 'context.only':
+                                    ctx.addFailureAt(node.getStart(), text.length, Rule.FAILURE_STRING_CONTEXT);
+                                    break;
+
+                                default: // required per tslint rules for switch statements.
```

Comment:
Intentionally went with the `switch` statement over the `if/elseif` for clarity. The forced inclusion of `default` mars it, admittedly. I can switch to the `if/else if` if required.

Reply:
+1 for getting rid of continuous `.getText()` calls!